Sparkle/SPUUpdater.m

```diff
@@ -201,30 +198,44 @@ - (BOOL)checkIfConfiguredProperly:(NSError * __autoreleasing *)error
         return NO;
     }
     
-    BOOL servingOverHttps = [[[feedURL scheme] lowercaseString] isEqualToString:@"https"];
-    if (!servingOverHttps) {
-        BOOL foundXPCPersistentDownloaderService = NO;
-        BOOL foundATSPersistentIssue = [self checkATSIssueForBundle:SPUXPCServiceBundle(@DOWNLOADER_BUNDLE_ID) getBundleExists:&foundXPCPersistentDownloaderService];
-        
-        NSBundle *mainBundle = [NSBundle mainBundle];
-        BOOL foundATSMainBundleIssue = NO;
-        if (!foundATSPersistentIssue && !foundXPCPersistentDownloaderService) {
-            BOOL foundATSIssue = ([mainBundle objectForInfoDictionaryKey:@"NSAppTransportSecurity"] == nil);
-            BOOL updatingMainBundle = [self.host.bundle isEqualTo:mainBundle];
-            
-            if (updatingMainBundle) {
-                // The only way we'll know for sure if there is an issue is if the main bundle is the same as the one we're updating
-                // We don't want to generate false positives..
-                foundATSMainBundleIssue = foundATSIssue;
+    BOOL servingOverHttps = NO;
+    NSError *feedError = nil;
+    NSURL *feedURL = [self retrieveFeedURL:&feedError];
+    if (feedURL == nil) {
+        if (requireFeedURL) {
+            if (error != NULL) {
+                *error = feedError;
             }
+            return NO;
         }
-        
-        if (foundATSPersistentIssue || foundATSMainBundleIssue) {
-            if (!self.loggedATSWarning) {
-                // Just log a warning. Don't outright fail in case we are wrong (eg: app is linked on an old SDK where ATS doesn't take effect)
-                SULog(SULogLevelDefault, @"The feed URL (%@) may need to change to use HTTPS.\nFor more information: https://sparkle-project.org/documentation/app-transport-security", [feedURL absoluteString]);
+    }
+    
+    if (feedURL != nil) {
+        servingOverHttps = [[[feedURL scheme] lowercaseString] isEqualToString:@"https"];
+        if (!servingOverHttps) {
+            BOOL foundXPCPersistentDownloaderService = NO;
+            BOOL foundATSPersistentIssue = [self checkATSIssueForBundle:SPUXPCServiceBundle(@DOWNLOADER_BUNDLE_ID) getBundleExists:&foundXPCPersistentDownloaderService];
+            
+            NSBundle *mainBundle = [NSBundle mainBundle];
+            BOOL foundATSMainBundleIssue = NO;
```

Comment:
Maybe just drop support for ATS exceptions and require HTTPS all the time?

Reply:
We may be able to simplify this and remove the ATS checks and just say feed may need to be https. I don't think it makes sense to always require https as the system prods that (it makes more sense to require (Ed)DSA imo). The test app uses http too.